exercises/exercise_2/README.md

```diff
@@ -1,219 +1,195 @@
-web          1        33000 dockerdemos/lab-web
-words        3              dockerdemos/lab-words
-```
-
-`inspect` command will
-* pretty print the metadata
-* list the services
-* but also volumes
-* networks
-* secrets 
-* attachments
-* parameters
-
-**NOTE**: All `docker-app` commands have multiple ways to take an application as an argument:
-```sh
-# You can omit it if current directory is a `*.dockerapp`
-myapp.dockerapp $ docker-app inspect
-# or if there is only one `*.dockerapp` (file or directory) in the current directory
-words $ docker-app inspect
-# you can reference it by its name only `path/to/myapp`
-workshop $ docker-app inspect words/myapp
-# or by its full path `path/to/myapp.dockerapp`
-workshop $ docker-app inspect words/myapp.dockerapp
-```
-
-* **Try** to inspect the application from different working directories.
-
-## Validate your application
-
-The `validate` command checks everything is ok with your application:
-* Compose file is valid
-* Required Metadata are filled (name, version) and have the good type
-* Default parameters
-
-* **Validate** your application
-```sh
-words $ docker-app validate myapp
-words $ echo $?
-0
-```
-
-Let's modify a bit this app:
-* **Comment** the version field in `myapp.dockerapp/metadata.yml` using `#`
-* **Add** some garbage characters to the name: `name: myapp$%%^`
-* **Validate** the application
-```sh
-words $ docker-app validate myapp
-Error: failed to validate metadata:
-- name: Does not match format 'hostname'
-- version: version is required
-words $ echo $?
-1
-```
-* **Fix** `myapp.dockerapp/metadata.yml`
-
-* **Comment** the `services:` line in `myapp.dockerapp/docker-compose.yml`
-* **Validate** the application
-```sh
-words $ docker-app validate myapp
-Error: failed to load composefiles: failed to parse Compose file version: "3.7"
-#services:
-    web:
-     image: dockerdemos/lab-web
-     ports:
-      - "33000:80"
-
-    words:
-     image: dockerdemos/lab-words
-     deploy:
-       replicas: 3
-       endpoint_mode: dnsrr
-
-    db:
-     image: dockerdemos/lab-db
-
-: yaml: line 2: did not find expected key
-
-words $ echo $?
-1
-```
-* **Fix** `myapp.dockerapp/docker-compose.yml`
-
-**Summary**
-* `docker-app` let's you define metadata and parameters on top of a compose file
-* `inspect` displays all important informations in a compact way
-* `validate` checks everything is ok in the application. You can use it as a CI step.
-* `split`/`merge` transforms your application to a multiple-files/one-file. The one-file version is easy to share.
+## Exercise Objective
+
+By the end of this exercise, you will have:
+
+- Created a Docker Application package
+- Learned about the application package components
+- Learned about merging and splitting the components
+- Learned how to inspect and validate the application
+
+
+
+## Docker Application Overview
+
+An application package is a set of 3 documents, plus custom files called `attachments`:
```

Comment:
Do you think we can keep this part, or at least rephrased ?

> `Application Package` is a construction above `Compose file` to improve the application lifecycle and workflow, from development to test to production.

I like to introduce `docker app` as a construction on top of the `Compose file`, so people won't see it as "yet another format" to learn.